plugin/src/main/java/org/opensearch/ml/model/MLModelAndNodeManager.java

```diff
@@ -0,0 +1,294 @@
+        for (SearchHit hit : hits) {
+            try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, hit.getSourceRef())) {
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                MLTask mlTask = MLTask.parse(parser);
+
+                String workerNode = mlTask.getWorkerNode();
+                MLTaskType mlTaskType = mlTask.getTaskType();
+                MLTaskState mlTaskState = mlTask.getState();
+
+                if (workerNode.contains(nodeId) && mlTaskType == MLTaskType.LOAD_MODEL && mlTaskState == MLTaskState.COMPLETED) {
+                    autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
+                }
+            } catch (Exception e) {
+                reTryTimes++;
+                // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+                saveLatestReTryTimes(nodeId, reTryTimes);
+                log.error("ML Task id {} failed to parse, the reason is: {}", hit.getId(), e);
+                throw new RuntimeException(e);
+            }
+        }
+
+        // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+        saveLatestReTryTimes(nodeId, reTryTimes);
+    }
+
+    /**
+     *
+     * @param nodeId
+     * @param modelId
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeAndModelId(String nodeId, String modelId) {
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, new String[] { nodeId }, false, false);
+        LoadModelResponse loadModelResponse = client.execute(MLLoadModelAction.INSTANCE, mlLoadModelRequest).actionGet();
+
+        if (loadModelResponse == null) {
+            throw new RuntimeException("Can't auto reload model " + modelId + " by node " + nodeId);
+        }
+    }
+
+    @VisibleForTesting
+    Integer getReTryTimes(String nodeId) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MODEL_LOAD_RETRY_TIMES_FIELD }, null);
+        QueryBuilder queryBuilder = new TermQueryBuilder(NODE_ID_FIELD, nodeId);
+        searchSourceBuilder.query(queryBuilder);
+        SearchRequest searchRequest = new SearchRequest().source(searchSourceBuilder).indices(ML_MODEL_RELOAD_INDEX);
+        SearchResponse response = client.execute(MLModelSearchAction.INSTANCE, searchRequest).actionGet(5000);
+
+        if (response == null) {
+            throw new RuntimeException("can't get retry times by " + nodeId);
+        }
+        SearchHits searchHits = response.getHits();
+        if (searchHits == null) {
+            throw new RuntimeException("can't get retry times by " + nodeId);
+        }
+        SearchHit[] hits = searchHits.getHits();
+        if (CollectionUtils.isEmpty(hits)) {
+            throw new RuntimeException("can't get retry times by " + nodeId);
```

Comment:
If no record exist, we should treat the retry time as 0 instead of throw exception